api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -298,15 +298,14 @@ private void streamToOutput(final boolean streamTimeoutReached) {
                         currentTimeMillis,
                         Math.min(getBatchLimitEvents(), messagesAllowedForPartition),
                         getParameters().batchTimeoutMillis,
-                        streamTimeoutReached);
-
+                        streamTimeoutReached,
+                        inDlqMode(partition));
```

Comment:
It looks more promising to make the decision in `getMessagesAllowedToSend()`, by passing the current partition there.  It should then return either 1 or 0 for any partition that is currently in the auto-DLQ search mode.  Wdyt?

Reply:
moved logic to `getMessagesAllowedForPartition`